src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -32,30 +33,53 @@
 import ch.njol.util.coll.CollectionUtils;
 import ch.njol.yggdrasil.YggdrasilSerializable;
 
-/**
- * @author Peter Güttinger
- * @edited by Mirreducki. Increased maximum timespan.
- */
 public class Timespan implements YggdrasilSerializable, Comparable<Timespan> { // REMIND unit
 
-	private final static Noun m_tick = new Noun("time.tick");
-	private final static Noun m_second = new Noun("time.second");
-	private final static Noun m_minute = new Noun("time.minute");
-	private final static Noun m_hour = new Noun("time.hour");
-	private final static Noun m_day = new Noun("time.day");
-	private final static Noun m_week = new Noun("time.week");
-	private final static Noun m_month = new Noun("time.month");
-	private final static Noun m_year = new Noun("time.year");
-	final static Noun[] names = {m_tick, m_second, m_minute, m_hour, m_day, m_week, m_month, m_year};
-	final static long[] times = {50L, 1000L, 1000L * 60L, 1000L * 60L * 60L, 1000L * 60L * 60L * 24L,  1000L * 60L * 60L * 24L * 7L,  1000L * 60L * 60L * 24L * 30L,  1000L * 60L * 60L * 24L * 365L};
-	final static HashMap<String, Long> parseValues = new HashMap<>();
+
+	private enum Times {
+
+		TICK(new Noun("time.tick"), 50L),
+		SECOND(new Noun("time.second"), 1000L),
+		MINUTE(new Noun("time.minute"), SECOND.time * 60L),
+		HOUR(new Noun("time.hour"), MINUTE.time * 60L),
+		DAY(new Noun("time.day"), HOUR.time * 24L),
+		WEEK(new Noun("time.week"), DAY.time * 7L),
+		MONTH(new Noun("time.month"), DAY.time * 30L),
+		YEAR(new Noun("time.year"), DAY.time * 365L);
+
+		private final Noun name;
+		private final long time;
+
+		Times(Noun name, long time) {
+			this.name = name;
+			this.time = time;
+		}
+
+	}
+
+	public static final Pattern TIMESPAN_PATTERN = Pattern.compile("^\\d+:\\d\\d(:\\d\\d){0,2}(\\.\\d{1,4})?$");
+	public static final Pattern TIMESPAN_NUMBER_PATTERN = Pattern.compile("^\\d+(\\.\\d+)?$");
+	private static final HashMap<String, Long> parseValues = new HashMap<>();
+	private final long millis;
```

Comment:
You should put this field below the static stuff, where it originally was. It's better to keep that separation between static and non-static